ext/soap/soap.c

```diff
@@ -995,6 +1001,12 @@ PHP_METHOD(SoapServer, setPersistence)
 	if (service->type == SOAP_CLASS) {
 		if (value == SOAP_PERSISTENCE_SESSION ||
 			value == SOAP_PERSISTENCE_REQUEST) {
+			if (value == SOAP_PERSISTENCE_SESSION && !zend_hash_str_exists(&module_registry, "session", sizeof("session")-1)) {
```

Comment:
probably stupid question, do not know much soap, but is it possible to know if session is supported once and earlier ?

Reply:
I can check that at module startup, not sure if it's worth doing.

Reply:
yeah just nitpicking no worries :)

Reply:
I thought about it. This method is normally only called once. So caching the result of the lookup at startup doesn't rlly make sense.